src/main/java/com/example/LongestPartStartIndexFinder.java

```diff
@@ -0,0 +1,70 @@
+package com.example;
+
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+public class LongestPartStartIndexFinder implements Collector<Integer, Integer, Integer> {
```

Comment:
What is the benefit of inheriting from Collector?  Simple one static method should be enough :)
I see you have to implement many methods which generates noise.
Related clean code principle: Single responsibility principl

Reply:
Honestly nothing, it was a little journey. It is a demonstration, it is not good way to use collectors for primitive types and this solution has bad performance like as the other one.